frontend-react/src/hooks/UseOktaMemberships.ts

```diff
@@ -96,46 +98,33 @@ export const getSettingsFromOrganization = (
     };
 };
 
-export const makeMembershipMapFromToken = (
-    token: AccessToken
-): Map<string, MembershipSettings> => {
-    // Extracts claims from token
-    const organizationClaim = getOktaGroups(token);
-    const settings: Map<string, MembershipSettings> = new Map();
-    // Creates map from claims
-    organizationClaim.forEach((org: string) => {
-        settings.set(org, getSettingsFromOrganization(org));
-    });
-    return settings;
-};
-
 const defaultState: MembershipState = {
     // note that active will be set to {} rather than undefined in most real world cases on initialization
     // see `calculateMembershipsWithOverride` for logic
     activeMembership: null,
-    memberships: undefined,
     initialized: false,
 };
 
 export const membershipsFromToken = (
     token: AccessToken | undefined
 ): Partial<MembershipState> => {
-    // One big undefined check to see if we have what we need for the next line
+    // Check if we even have claims
     if (!token?.claims) {
         return omit(defaultState, "initialized");
     }
-    const claimData: Map<string, MembershipSettings> =
-        makeMembershipMapFromToken(token);
-    // Catch anyone with no claim data
-    if (!claimData.size) {
+    const claims = toRSClaims(token.claims);
+    // Check if we have any organization claims
+    if (!claims?.organization?.length) {
         return omit(defaultState, "initialized");
     }
-    // Get defaults
-    const [first] = claimData.keys();
-    const active = claimData.get(first);
+    // If we're admins, set that always, else set default to first in org claim array
+    const adminIndex = claims.organization.indexOf(PRIME_ADMINS);
```

Comment:
Smalllll point, but we could get rid of the index middleman and just return it directly:

```js
return {
    activeMembership: claims.organization.find((org) => org === PRIME_ADMINS) || claims.organization[0]
}
```

Reply:
Niiiice, love these little in-line tricks! Patched up